gloo/transport/tcp/listener.h

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <list>
+#include <unordered_map>
+
+#include <gloo/transport/tcp/address.h>
+#include <gloo/transport/tcp/attr.h>
+#include <gloo/transport/tcp/error.h>
+#include <gloo/transport/tcp/loop.h>
+#include <gloo/transport/tcp/socket.h>
+
+namespace gloo {
+namespace transport {
+namespace tcp {
+
+// Listener deals with incoming connections. Incoming connections
+// write a few bytes containing a sequence number. This sequence
+// number is read off the socket and matched to a local sequence
+// number. If there is a match, the socket is passed to the
+// corresponding pair. If it can't be matched, it is stashed until a
+// pair with the sequence number calls `waitForConnection`.
+class Listener final : public Handler {
+ public:
+  using connect_callback_t =
+      std::function<void(std::shared_ptr<Socket> socket, Error error)>;
+
+  static constexpr auto kBacklog = 512;
+
+  explicit Listener(std::shared_ptr<Loop> loop, const attr& attr);
```

Comment:
nit: do we use explicit on multi-arg constructors?

Reply:
Good catch. Fixed.